src/nimblepkg/options.nim

```diff
@@ -268,8 +278,18 @@ proc parseArgument*(key: string, result: var Options) =
     result.showHelp = true
   of actionCustom:
     result.action.arguments.add(key)
-  else:
-    discard
+  of actionSample:
+    if result.action.package == "":
+      result.action.package = key
+    elif result.action.sample == "" and 
+      (not ('.' in key)) and (key != "#head"):
+        result.action.sample = key
+    elif (not ('.' in key)) and (key != "#head"):
+      result.action.ver = Version(result.action.sample)
+      result.action.sample = key
+    else:
+      result.action.ver =  Version(key)
```

Comment:
yeah, please don't do this. Reuse the logic for parsing versions that I have for `install`.

Reply:
done.